### PR TITLE
Setup basic testing framework

### PR DESCRIPTION
**Pull request by kjmerf**
Purpose of pull request is to add linting capability to repo. All you need to do is run:

```
pip install pytest-flake8
./test.sh
```
And the linting test will run against the python files in the datascroller/datascroller directory.

My hope is that the basic setup can be used as a framework for more advanced testing.

### CODE REVIEW

**Comment by baogorek** [comment]
Good stuff, @kjmerf. I've yet to use linting like this in a Python package. Will dig in as soon as I can and am looking forward to the learning.

**Comment by baogorek** [comment]
@kjmerf , one question about the name of the file `test.sh`. Is your plan for this script wider in scope than linting? If not, I wonder if `run_linter.sh` would be a better name. We also need unit tests and they will be in the tests/ directory. Perhaps this file should just lint, or lint and run the unit tests. What are your thoughts on that?

**Comment by kjmerf** [comment]
@baogorek personally I like a "run all the tests" script, which is my intention here. I did a bit more research and looks like we can incorporate linting into a more generic pytest framework. I set up a really basic test for the demo function to show you what I mean. Have a look and let me know what you think.

**Comment by baogorek** [comment]
@kjmerf ok I'm with you; let's keep it the way it is. I'm a little behind today but I'm looking forward to trying out your script. We'll get this merged by end of week for sure.

**Comment by kjmerf** [comment]
Sounds good

File: test/test_demo.py
Comment (baogorek) [inline test/test_demo.py:9]: Even thought it's pretty obvious, I'd put a """docstring""" in the one blank line with a quick (less than 79 char) statement about the function.

File: test/test_demo.py
Comment (baogorek) [inline test/test_demo.py:2]: I was about to suggest knocking out the blank lines in the import, but then I saw that PEP8 recommends a blank line between standard library, third party, and local libraries. I learned something new!

File: setup.py
Comment (baogorek) [inline setup.py:20]: Later on down the line we can try to get fancier with `test_requires` as shown [here](https://python-packaging.readthedocs.io/en/latest/testing.html). I am perfectly fine with how it is currently for this iteration.

File: datascroller/command_line.py
Comment (baogorek) [inline datascroller/command_line.py:10]: If you want, you can knock out this comment. We're not using non-blocking mode in the larger module and commented code in general is not good. Optional for this PR.

File: datascroller/scroller.py
Comment (baogorek) [inline datascroller/scroller.py:63]: Looks like I left some stray comments in here as well. Optional to take care of now.

File: datascroller/scroller.py
Comment (baogorek) [inline datascroller/scroller.py:284]: I also don't think this comments doing anything for me or anybody else. Feel free to delete. Optional

File: datascroller/scroller.py
Comment (baogorek) [inline datascroller/scroller.py:299]: A line I used to debug. Can be removed.

**Review by baogorek — APPROVED** [review]
@kjmerf I tried out your linting script and it worked great. I did leave a few last comments and optional changes but this PR is ready to merge as it is.

**Comment by kjmerf** [comment]
@baogorek I submitted a new commit to KevinDev but the PR did not update. Instead I got an "outdated" message. So I went ahead and merged the branch as-is. Couple questions though: do we have a policy in place to freeze PRs once submitted instead of auto-updating them given new commits in the source branch? Secondly, if you approve the PR and I submit a new commit, do you need to approve again?

**Comment by baogorek** [comment]
Congrats on your first merge @kjmerf! I see the number "3" in the contributors box, which is excellent.

About the "out of date" message, John's 1-liner PR got merged in before this one so you would have had to pull the updated master into your branch before pushing again. Since this PR came from your own forked repo, it's [a little more complicated](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/merging-an-upstream-repository-into-your-fork). You now have push access to this repository so anytime you want you can reclone (or change the [remote URL](https://help.github.com/en/github/using-git/changing-a-remotes-url)), that will give you the ability to create a new branch and push right to this repo (on any branch that's not master).

I'm fine with being relatively informal about the Pull Request approvals. If you get an approval with a request to take care of a couple more minor things, you can use your judgement about when someone should look at it again. If you deleted the extra whitespace as was requested but got inspired and added a new feature, of course that'd be a time to ask for a new review.

Glad you're on the project, Kevin,
Ben

**Pull request merged**
